packages/sui-cz/PrompterManager.js

```diff
@@ -0,0 +1,188 @@
+/* eslint-disable no-console */
+
+const fs = require('fs')
+const path = require('path')
+const { exec } = require('child_process')
+const colors = require('colors')
+const editor = require('editor')
+const temp = require('temp').track()
+const ErrorCommitSaver = require('./ErrorCommitSaver')
+const buildCommit = require('./buildCommit')
+const config = require('./config')
+const packagesDir = path.join(process.cwd(), config.getPackagesFolder())
+const scopes = config.getScopes().sort().map(name => ({ name }))
+const commitTypes = require('./commitTypes').types
+
+const typesWithOtherScopes = ['feat', 'fix', 'release', 'test', 'docs', 'chore']
+const otherScopes = [{ name: 'META' }, { name: 'examples' }]
+const allowedBreakingChanges = ['feat', 'fix']
+const getCommitTypesMapped = () => Object.keys(commitTypes).map((value) => ({ value, name: commitTypes[value].description }))
+
+/**
+ * Create the prompter manager, is the one that have the responsibility of manage different prompt flows
+ * class
+ */
+class PrompterManager {
+  /**
+   * Init the edit flow.
+   * The edit flow will save our commit on a temporally file, open it on an editor and wait expecting for changes.
+   * @return {undefined}
+   */
+  static startEditFlow (answers) {
+    temp.open(null, (err, info) => {
+      if (!err) {
+        fs.write(info.fd, buildCommit(answers), () => {})
+        fs.close(info.fd, () => {
+          editor(info.path, (code) => {
+            if (!code) {
+              var commitStr = fs.readFileSync(info.path, {
+                encoding: 'utf8'
+              }, () => {})
+              this.doCommit(commitStr)
+            } else {
+              console.log(`Editor returned non zero value. Commit message was:\n ${buildCommit(answers)}`)
+            }
+          })
+        })
+      }
+    })
+  }
+
+  /**
+   * Check if modified files are present
+   * @param  {[type]}  path Folder to check
+   * @return {Promise<Boolean>}
+   */
+  static checkIfHasChangedFiles (path) {
+    return new Promise((resolve, reject) => {
+      exec(`git status ${path}`, { cwd: path }, (err, output) => {
+        err ? reject(err) : resolve(!output.includes('nothing to commit'))
+      })
+    })
+  }
+
+  /**
+   * The doCommit method will init an error listener, start the git commit flow, and if all goes ok discard the old commit if was saved before.
+   * @param {string} commitString
+   */
+  static doCommit (commitString) {
+    ErrorCommitSaver.initErrorListener(commitString)
+    this.commit(commitString)
+    ErrorCommitSaver.discardOldCommit()
+  }
+
+  /**
+   * The get commmit steps method will return an array of objects that corresponds to the different steps of our commit promptet.
+   * [Step1, ...2, ...3, ...4, ...5, ...6, ...7]
+   * @returns {[Object,Object,Object,Object,Object,Object,Object]}
```

Comment:
not important but this looks really weird.

This would normally be: `[]Object` or `Array<Object>` or similar
You can clarify that is an array of steps if you feel it's needed

Reply:
Totally agree 🙃  Sorry about that